File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/file_encryption_decryption/File/ExcelFile.java
package gateway.hitrontech.com.encryption.fragment.file_encryption_decryption.File;

import gateway.hitrontech.com.encryption.bean.EncryptionBean;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Iterator;
import org.apache.poi.ss.usermodel.Cell;
import org.apache.poi.ss.usermodel.Row;
import org.apache.poi.ss.usermodel.Sheet;
import org.apache.poi.ss.usermodel.Workbook;
import org.apache.poi.xssf.usermodel.XSSFWorkbook;

public class ExcelFile implements FileImpl {

  @Override
  public void readFile(ArrayList<EncryptionBean> list, String filePath) throws IOException {
    File file = new File(filePath);
    if (file.exists()) {
      try {
        FileInputStream fileInputStream = new FileInputStream(file);
        Workbook workbook = new XSSFWorkbook(fileInputStream);
        Sheet sheet = workbook.getSheetAt(0);

        for (Row currentRow : sheet) {
          Iterator<Cell> cellIterator = currentRow.iterator();
          EncryptionBean tmp = new EncryptionBean();
          Cell plainCell = cellIterator.next();
          Cell keyCell = cellIterator.next();
          if (null == plainCell || null == keyCell) {
            continue;
          } else {
            plainCell.setCellType(Cell.CELL_TYPE_STRING);
            keyCell.setCellType(Cell.CELL_TYPE_STRING);
          }
          tmp.setPlainText(plainCell.getStringCellValue());
          tmp.setKey(keyCell.getStringCellValue());
          list.add(tmp);
        }

        fileInputStream.close();
      } catch (IOException e) {
        e.printStackTrace();
      }
    } else {
      throw new IOException();
    }
  }

  private Cell setCellValueType(Cell cell) {
    cell.setCellType(Cell.CELL_TYPE_STRING);
    return cell;
  }

  @Override
  public void writeFile(ArrayList<EncryptionBean> list, String filePath) {
    try {
      Workbook workbook = new XSSFWorkbook();
      Sheet sheet = workbook.createSheet("TARGET");

      int rowNumber = 0;

      for (EncryptionBean bean : list) {
        Row row = sheet.createRow(rowNumber++);
        Cell first = row.createCell(0);
        first.setCellValue(bean.getPlainText());

        Cell second = row.createCell(1);
        second.setCellValue(bean.getKey());
        Cell third = row.createCell(2);
        third.setCellValue(bean.getCipherText());
      }

      FileOutputStream fileOutputStream = new FileOutputStream(filePath);
      if (!list.isEmpty()) {
        sheet.setColumnWidth(0, 50 * 255);
        sheet.setColumnWidth(1,
            (list.get(0).getKey().length() > 255 ? 255 : list.get(0).getKey().length())
                * 255);
        sheet.setColumnWidth(2, 50 * 255);
      }
      workbook.write(fileOutputStream);
      workbook.cloneSheet(0);
      fileOutputStream.close();
    } catch (IOException e) {
      e.printStackTrace();
    }
  }
}

File: app/src/main/java/gateway/hitrontech/com/encryption/fragment/file_encryption_decryption/File/ExcelFile.kt
package gateway.hitrontech.com.encryption.fragment.file_encryption_decryption.File

import gateway.hitrontech.com.encryption.bean.EncryptionBean
import org.apache.poi.hssf.usermodel.HSSFWorkbook
import org.apache.poi.ss.usermodel.IndexedColors
import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException
import java.util.*

class ExcelFile : FileImpl {

    override fun readFile(list: ArrayList<EncryptionBean>, filePath: String) {
        val file = File(filePath)
        if (file.exists()) {
            try {
                val fileInputStream = FileInputStream(file)
                val workbook = HSSFWorkbook(fileInputStream)
                val sheet = workbook.getSheetAt(0)
                val iterator = sheet.iterator()

                var count = 0
                while (iterator.hasNext()) {
                    val currentRow = iterator.next()
                    if (count != 0) {
                        val cellIterator = currentRow.iterator()
                        val tmp = EncryptionBean()
                        tmp.plainText = cellIterator.next().stringCellValue
                        tmp.key = cellIterator.next().stringCellValue
                        list.add(tmp)
                    }
                    count++
                }

                fileInputStream.close()
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }
    }

    override fun writeFile(list: ArrayList<EncryptionBean>, filePath: String) {
        try {
            val workbook = HSSFWorkbook()
            val sheet = workbook.createSheet("TARGET")

            var rowNumber = 0
            val font = workbook.createFont()
            font.boldweight = 16.toShort()
            font.fontHeight = 20.toShort()
            font.color = IndexedColors.RED.getIndex()
            val header = sheet.createRow(rowNumber++)
            val style = workbook.createCellStyle()
            style.setFont(font)
            header.setRowStyle(style)
            val plainText = header.createCell(0)
            plainText.setCellValue("PlainText")
            val key = header.createCell(2)
            key.setCellValue("KEY")
            val cipherText = header.createCell(4)
            cipherText.setCellValue("CipherText")

            for (bean in list) {
                val row = sheet.createRow(rowNumber++)
                val first = row.createCell(0)
                first.setCellValue(bean.plainText)
                val second = row.createCell(2)
                second.setCellValue(bean.key)
                val third = row.createCell(4)
                third.setCellValue(bean.cipherText)
            }

            val fileOutputStream = FileOutputStream(filePath)
            workbook.write(fileOutputStream)
            workbook.cloneSheet(0)

        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}
